src/main/java/org/sagebionetworks/web/client/widget/team/JoinTeamWidget.java

```diff
@@ -107,7 +112,52 @@ public void configure(String teamId, boolean isChallengeSignup, TeamMembershipSt
 		this.isMemberMessage = isMemberMessage;
 		this.successMessage = successMessage;
 		this.buttonText = buttonText;
-		view.configure(authenticationController.isLoggedIn(), teamMembershipStatus, isMemberMessage, buttonText, requestOpenInfoText, isSimpleRequestButton);
+		this.teamMembershipStatus = teamMembershipStatus;
+		view.clear();
+		if (buttonText != null && !buttonText.isEmpty()) {
```

Comment:
This is a direct migration from the view.

If you don't have buttonText/requestOpenText/isMemberText, then leave at the default, already set values.